### ISSUE TITLE
Why multiple ports per device?

### ISSUE THREAD

**Issue by sole** [mozilla/node-firefox-forward-ports#2]
You're already trying to reestablish existing connections, so why do we expect multiple port forwards?

Do we know of any scenario where this is happening? Can it even work?

I am writing some code using this module and it is cumbersome to have to access the port by index. Instead of being able to do

``` javascript
connect(device.port);
```

I have to do this?

``` javascript
connect(device.ports[0].port);
```

or even check if ports has `length > 0`, etc

Can I suggest changing this to only use one port or am I missing something here? pinging @lmorchard

**Comment by lmorchard** [comment]
[FWIW, this came up in comments on the initial PR](https://github.com/mozilla/node-firefox-forward-ports/pull/1#discussion_r26846305). (Link for context)

TL;DR: While _this_ module tries to reuse existing forwards, that doesn't mean that something else hasn't made them. There _can_ be multiple local port forwards, even to the same remote port - e.g. [the debugging service got 2 manual forwards](https://github.com/mozilla/node-firefox-forward-ports/pull/1#discussion_r26847898), possibly by mistake. 

So, we could just report one, despite there being multiple. Maybe that could be an option flag to just report the first one found? Could also make that true by default for convenience's sake. (Or maybe instead, an advanced "fullPorts" option that includes the extra ports & details but is false by default.)

You also may eventually _want_ forwards to multiple remote ports - e.g. remote debugging and marionette test services. We don't do anything with marionette right now, but that will probably change.

**Comment by lmorchard** [comment]
Oh, one other thing: I'm working today on adding port forward detection to node-firefox-find-ports. That will probably be more convenient as input to `connect()`. But, we might want a flag there to just use the first debugging port found per device, as well.

**Comment by lmorchard** [comment]
Okay, [got a first attempt at that added functionality in node-firefox-find-ports](https://github.com/mozilla/node-firefox-find-ports/pull/34). That will ignore multiple ports per device, by default. The way I envision this stuff being used is something like this crazy example:

``` javascript
var appPath = './node-firefox-install-app/examples/sampleApp';

Promise.all([

  // Find the devices and forward ALL the ports
  findDevices().then(forwardPorts),

  // Find and start ALL the simulators
  findSimulators().then(function (simulators) {
    // Hmm, this could probably be simpler if the output of findSimulator
    // matched the input of startSimulator
    return Promise.all(simulators.map(function (simulator) {
      return startSimulator({
        binary: simulator.bin,
        profile: simulator.profile
      });
    }));
  })

])

// Find the ports for devices and simulators, ignoring the results of the above
.then(function () { return findPorts(); })

// Install and launch the app on all runtimes
.then(function (ports) {
  return Promise.all(ports.map(function (runtime) {
    return connect(runtime.port).then(function (client) {
      return installApp({
        client: client,
        appPath: appPath
      }).then(function (installedApp) {
        return launchApp({
          client: client, 
          manifestURL: installedApp.manifestURL 
        });
      });
    });
  });
});
```

(That's totally just off the top of my head - it's probably broken in some way!)
